internal/daemon/controller/common/scopeids/scope_ids.go

```diff
@@ -353,3 +353,143 @@ func GetListingScopeIds(
 	}
 	return scopeResourceInfo.ScopeIds, scopeMap, nil
 }
+
+// AuthorizedForList retrieves and returns all scopes where a user is authorized
+// to perform a *list* action on. It looks recursively from `rootScopeId`.
+func AuthorizedForList(
+	ctx context.Context, // The context to use when listing in the DB, if required
+	iamRepoFn common.IamRepoFactory, // An IAM repo function to use for a scope listing call
+	authResults auth.VerifyResults, // The original auth results from the list command
```

Comment:
I am wondering if this function should just be a method on `auth.VerifyResults`. I don't know if that causes some import issues, but I think it would work since, an `auth.VerifyResults` already has a `common.IamRepoFactory` via the [verifier](https://github.com/hashicorp/boundary/blob/a21ff8b00476fad4238e98295673a70ca59e3532/internal/daemon/controller/auth/auth.go#L94). That would turn this into something like:

```go
func (r *VerifyResults) ScopesAuthorizedForList(ctx context.Context, rootScopeId string, resourceType resource.Type) (map[string]*scopes.ScopeInfo, error)
```

Reply:
I shall give this a try!

Reply:
Works great, and it's actually more helpful because what we have in the service functions is a `VerifyResults` object - [771086a](https://github.com/hashicorp/boundary/pull/2336/commits/771086aaacde57756d84ea6bfe19f917ee4fb42c)